ethcore/sync/src/chain/handler.rs

```diff
@@ -811,6 +816,59 @@ impl SyncHandler {
         Ok(())
     }
 
+    /// Called when peer sends us a set of new pooled transactions
+    pub fn on_peer_new_pooled_transaction_hashes(
+        sync: &mut ChainSync,
+        io: &mut dyn SyncIo,
+        peer_id: PeerId,
+        tx_rlp: &Rlp,
+    ) -> Result<(), DownloaderImportError> {
+        for item in tx_rlp {
+            let hash = item
+                .as_val::<H256>()
+                .map_err(|_| DownloaderImportError::Invalid)?;
+
+            if io.chain().queued_transaction(hash).is_none() {
+                sync.peers
+                    .get_mut(&peer_id)
+                    .map(|peer| peer.unfetched_pooled_transactions.insert(hash));
```

Comment:
There should be some reasonable upper bound here.
1. We don't verify the peer speaks 65 at all, when we receive a package. So it may inform us about new hashes, but we never really check them.
2. They can keep spamming random hashes to us, and we will figure that we don't have them in the pool and keep extending this set.